test/gurka/test_avoids.cc

```diff
@@ -214,6 +217,50 @@ TEST_P(AvoidTest, TestAvoid2Polygons) {
   };
 }
 
+TEST_P(AvoidTest, TestAvoidShortcuts) {
+  auto node_b = avoid_map.nodes.at("B");
+  auto node_c = avoid_map.nodes.at("C");
+  auto node_e = avoid_map.nodes.at("E");
+  auto dx = std::abs(node_c.lng() - node_b.lng());
+  auto dy = std::abs(node_b.lat() - node_e.lat());
+
+  // create a small polygon on the shortcut: should fail if shortcuts are also avoided
+  //                x-x
+  //      A------B--|-|-C--F--G
+  //                x-x
+  //             |_________|
+  //               shortcut
+
+  // one clockwise ring
+  std::vector<ring_bg_t> rings;
+  rings.push_back({{node_b.lng() + 0.1 * dx, node_b.lat() + 0.01 * dy},
+                   {node_c.lng() - 0.1 * dx, node_c.lat() + 0.01 * dy},
+                   {node_c.lng() - 0.1 * dx, node_c.lat() - 0.01 * dy},
+                   {node_b.lng() + 0.1 * dx, node_b.lat() - 0.01 * dy},
+                   {node_b.lng() + 0.1 * dx, node_b.lat() + 0.01 * dy}});
+
+  // build request manually for now
+  auto lls = {avoid_map.nodes["A"], avoid_map.nodes["G"]};
+
+  // TODO: remove before commit
+  std::cout << gurka::dump_geojson_graph(avoid_map) << std::endl;
+
+  rapidjson::Document doc;
+  doc.SetObject();
+  auto& allocator = doc.GetAllocator();
+  auto value = get_avoid_polys(rings, allocator);
+  auto type = "avoid_polygons";
+  auto req = build_local_req(doc, allocator, lls, GetParam(), value, type);
+
+  // make sure the right exception is thrown
+  try {
+    gurka::do_action(Options::route, avoid_map, req);
```

Comment:
something approximately like:
```suggestion
    auto api = gurka::do_action(Options::route, avoid_map, req);
    size_t shortcut_count = 0;
    const auto& avoid_edges = api.options().costing_options(costing::type::auto).avoid_edges();
    for (const auto& avoid : avoid_edges){
      shortcut_count += graphreader.directededge(GraphId(avoid.id()))->shortcut();
    }
    ASSERT_TRUE(shortcut_count == 2);
    ASSERT_TRUE(avoid_edges.Size() == 4);
```